eclair-core/src/main/scala/fr/acinq/eclair/channel/ChannelTypes.scala

```diff
@@ -244,13 +244,15 @@ object ChannelOpenResponse {
       8888888P" d88P     888     888  d88P     888
  */
 
-sealed trait Data
+sealed trait Data {
+  def channelId: ByteVector32 = ByteVector32.Zeroes
```

Comment:
This shouldn't have a default value, that's too dangerous.
We must force all children to implement it explicitly (`Nothing` will be the only one returning `ByteVector32.Zeroes`).